samza-sql/src/main/java/org/apache/samza/sql/interfaces/SqlIOConfig.java

```diff
@@ -79,6 +82,14 @@ public SqlIOConfig(String systemName, String streamName, List<String> sourcePart
     Validate.notEmpty(samzaRelConverterName,
         String.format("%s is not set or empty for system %s", CFG_SAMZA_REL_CONVERTER, systemName));
 
+    if (isRemoteTable()) {
+      samzaRelTableKeyConverterName = streamConfigs.get(CFG_SAMZA_REL_TABLE_KEY_CONVERTER);
+      Validate.notEmpty(samzaRelTableKeyConverterName,
```

Comment:
Does all remote tables need a tableKeyConverters even if the table providers support primitive types which doesn't need a conversion?

Reply:
Based on the above discussion, The answer to this is Yes? We need a tableKeyConverter even for tables that just support primitives?

Reply:
Correct.